pkg/controller/ecs/service/setup.go

```diff
@@ -230,7 +230,10 @@ func preCreate(_ context.Context, cr *svcapitypes.Service, obj *svcsdk.CreateSer
 	obj.ClientToken = aws.String(string(cr.UID))
 	obj.Cluster = cr.Spec.ForProvider.Cluster
 	obj.LoadBalancers = generateLoadBalancers(cr)
-	obj.NetworkConfiguration = generateNetworkConfiguration(cr)
+	networkConfiguration := generateNetworkConfiguration(cr)
+	if networkConfiguration != nil {
+		obj.NetworkConfiguration = networkConfiguration
+	}
```

Comment:
I believe this can stay as it is, right? Or is there another place where the value should be set?

```suggestion
	obj.NetworkConfiguration = generateNetworkConfiguration(cr)
```